### PR TITLE
Centrality reco modification for 2018 heavy ion run

### PR DESCRIPTION
**Pull request by YeonjuGo**
fixed pixel collection name to be able to read them
and added hfecut, pixelplus, pixelminus, pixeltrackplus, pixeltrackminus variables
 : https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/181017_RecoModification_hf_pixel_variables_GO.pdf

updated pixel cut 
 :  https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/pixel-cluster-size-181018.pdf

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by YeonjuGo** [comment]
@slava77 
Please take a look at the presentation summarized modifications and related plots; https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/181017_RecoModification_hf_pixel_variables_GO.pdf
Thank you

File: RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc [hunk 1]
```diff
@@ -226,16 +228,20 @@ CentralityProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
...
 		 creco->etHFtowerSumMinus_ += tower.pt();
+         if(tower.energy() > 1.5) creco->etHFtowerSumECutMinus_ += tower.pt();
 		 if(eta < -hfEtaCut_) creco->etHFtruncatedMinus_ += tower.pt();
 	      }
 	   }else{
     	     if(reuseAny_){
 	      creco->etHFtowerSumMinus_ = inputCentrality->EtHFtowerSumMinus();
 	      creco->etHFtowerSumPlus_ = inputCentrality->EtHFtowerSumPlus();
+          creco->etHFtowerSumECutMinus_ = inputCentrality->EtHFtowerSumECutMinus();
+          creco->etHFtowerSumECutPlus_ = inputCentrality->EtHFtowerSumECutPlus();
```
Comment (slava77) [inline RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc]: please indent consistently

File: RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc [hunk 2]
```diff
@@ -319,12 +331,19 @@ CentralityProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
             if ( abeta > 2.5 && abeta < 5   && clusterSize < 9) continue;
 	  }
 	  nPixel++;
+      if(eta>=0) nPixel_plus++;
+      if(eta<0) nPixel_minus++;
```
Comment (slava77) [inline RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc]: indentation

File: RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc [hunk 3]
```diff
@@ -414,11 +433,24 @@ CentralityProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
     Handle<TrackCollection> pixeltracks;
     iEvent.getByToken(srcPixelTracks_,pixeltracks);
     int nPixelTracks = pixeltracks->size();
+    int nPixelTracksPlus = 0;
+    int nPixelTracksMinus = 0;
+
+    for(unsigned int i = 0 ; i < pixeltracks->size(); ++i){
```
Comment (slava77) [inline RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc]: ```for (auto const& track : pixelTracks)```

File: RecoHI/HiCentralityAlgos/python/HiCentrality_cfi.py [hunk 4]
```diff
@@ -38,7 +38,8 @@
 from Configuration.Eras.Modifier_pp_on_AA_2018_cff import pp_on_AA_2018
 for e in [pp_on_XeXe_2017, pp_on_AA_2018]:
     e.toModify(hiCentrality,
-               producePixelTracks = False,
+               producePixelTracks = True,
+               srcPixelTracks = cms.InputTag("hiConformalPixelTracks"),
                srcTracks = cms.InputTag("generalTracks"),
                srcVertex = cms.InputTag("offlinePrimaryVertices")
```
Comment (slava77) [inline RecoHI/HiCentralityAlgos/python/HiCentrality_cfi.py]: please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

**Comment by YeonjuGo** [comment]
@slava77 
I fixed things as you commented and also updated one more part. 
The new update is about pixel cut. We've been using this cut which is tuned for pre-phase 1 pixel detectors. So we'd like to update it. You can refer to the study on this: https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/pixel-cluster-size-181018.pdf
Thank you.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by fabiocos** [comment]
+1

**Pull request merged**